src/BenchmarkDotNet/Engines/CacheClearingStrategies/MemoryAllocator.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace BenchmarkDotNet.Engines.CacheClearingStrategies
+{
+    internal class MemoryAllocator : IMemoryAllocator
+    {
+        private const int HowManyMb = 14;
+        private const int SizeOfArray = HowManyMb * 1024 * 1024;
+        private readonly int[] memory = Enumerable.Range(1, SizeOfArray).ToArray();
+
+        public void AllocateMemory()
+        {
+            for (int i = 0; i < SizeOfArray; i++)
```

Comment:
We can use `i += 16` instead of `i++`, it will work faster with the same effect.